Bluetooth: controller: split: Port radio event abort

cvinayak opened the pull request:
Port the implementation that does radio event abort due to
ISR latencies. The implementation measures if the ISR could
not meet the hard real time deadline and closes the event
early.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: split: Port radio event abort

Port the implementation that does radio event abort due to
ISR latencies. The implementation measures if the ISR could
not meet the hard real time deadline and closes the event
early.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

carlescufi commented on subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c:
```diff
@@ -340,11 +341,25 @@ u32_t lll_evt_offset_get(struct evt_hdr *evt)
+	u32_t diff;
+
+	diff = ticker_ticks_diff_get(ticks_now, ticks_at_event);
+	diff += 3;
```
a small comment here about this hardcoded value?

cvinayak replied:
Help me here with a define name; 3 is the number of ticks between the high watermark and full tank. i.e. if the actual ticks plus 3 ticks exceeds the hard real-time deadline tick.

carlescufi replied:
`SAFETY_TICKS` or `TICKS_TO_DEADLINE`

cvinayak replied:
@carlescufi the right macro to use is HAL_TICKER_CNTR_CMP_OFFSET_MIN, hence updated now.

cvinayak commented:
@thoh-ot @aescolar ...if the changes seems ok (although its nordic lll specific), please do the needful.

cvinayak committed:
Bluetooth: controller: split: Add interface to get adv/scan handle

Added interface to get Advertiser and Scanner instance
handle for use in Lower Link Layer module.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

mtpr-ot reviewed (approved):
I don't see any issues here, but since this is mostly Nordic specific code (except for the getters), I'm not an expert on the impact. OK with me to merge, if accepted by Nordic.

aescolar commented:
@cvinayak As @carlescufi is out, would it make sense if instead @joerchan would take a look to have somebody else from Nordic giving their blessing? Otherwise just tell and I press the button.
